src/operations/operation.ts

```diff
@@ -121,6 +117,25 @@ export abstract class AbstractOperation<TResult = any> {
   }
 }
 
+/** @internal */
+export abstract class AbstractCallbackOperation<TResult = any> extends AbstractOperation {
```

Comment:
```suggestion
export abstract class AbstractCallbackOperation<TResult> extends AbstractOperation<TResult> {
```

Reply:
Why remove the default generic parameter?

Reply:
Because it's already defaulted in `AbstractOperation`.

Reply:
In that case I think we should keep it here and remove it from `AbstractOperation` because TypeScript doesn't understand that we have it defaulted for the subclasses of `AbstractCallbackOperation` that don't explicitly provide the parameter when they are declared so we're getting TypeScript errors.